### PR TITLE
avoid missing tiles when they not intersect 100% exactly perfect

### PR DESCRIPTION
**Pull request by jdroenner**

### CODE REVIEW

**Comment by jdroenner** [comment]
[bug in projektion](https://trello.com/c/Qk4Hqo4o/478-bug-in-projektion)

**Comment by jdroenner** [comment]
[bug in projektion](https://trello.com/c/Qk4Hqo4o/478-bug-in-projektion)

File: datatypes/src/operations/reproject.rs [hunk 1]
```diff
@@ -247,7 +247,7 @@ where
 {
     type Out = A;
     fn reproject_clipped(&self, projector: &P) -> Result<A> {
-        const POINTS_PER_LINE: i32 = 7;
+        const POINTS_PER_LINE: i32 = 21;
```
Comment (michaelmattig) [inline datatypes/src/operations/reproject.rs:250]: how did you decide on this number? Or was it just for debugging purposes? It doesn't influence the reproject_clipped or does it?

File: operators/src/adapters/raster_subquery_adapter.rs [hunk 2]
```diff
@@ -741,7 +741,8 @@ where
             .spatial_partition()
             .intersection(&query_rect.spatial_partition())
             .expect("should not be empty")
-            .reproject(&proj);
+            .reproject_clipped(&proj); // we need to use the fail tollerent reprojection here since there might be some pixels that are outside the valid bounds
```
Comment (michaelmattig) [inline operators/src/adapters/raster_subquery_adapter.rs:744]: can you add a test that fails without this change?

File: operators/src/adapters/raster_subquery_adapter.rs [hunk 3]
```diff
@@ -741,7 +741,8 @@ where
...
+                                       //TODO: there must be a better way for this
```
Comment (michaelmattig) [inline operators/src/adapters/raster_subquery_adapter.rs:745]: what do you dislike about this approach?

**Review by michaelmattig — CHANGES_REQUESTED** [review]

File: operators/src/processing/reprojection.rs [hunk 4]
```diff
@@ -1092,10 +1054,17 @@ mod tests {
...
             .unwrap();
 
-        qs.map(Result::unwrap)
+        let tiles = qs
+            .map(Result::unwrap)
             .collect::<Vec<RasterTile2D<u8>>>()
             .await;
 
+        // the test must generate 8x4 tiles
+        dbg!(tiles.len(), 32);
```
Comment (michaelmattig) [inline operators/src/processing/reprojection.rs:1063]: make it an assertion

File: operators/src/processing/reprojection.rs [hunk 5]
```diff
@@ -1092,10 +1054,17 @@ mod tests {
...
+
+        // none of the tiles should be empty
+        assert!(tiles.iter().all(|t| !t.is_empty()));
```
Comment (michaelmattig) [inline operators/src/processing/reprojection.rs:1066]: should we check the content for correctness also?

**Review by michaelmattig — CHANGES_REQUESTED** [review]

File: operators/src/processing/reprojection.rs (hunk 5 above)
Comment (jdroenner) [reply operators/src/processing/reprojection.rs:1066]: i guess we should do that with an other test

File: datatypes/src/operations/reproject.rs (hunk 1 above)
Comment (jdroenner) [reply datatypes/src/operations/reproject.rs]: i think 7 is to small but i changed it back

File: operators/src/processing/reprojection.rs (hunk 4 above)
Comment (jdroenner) [reply operators/src/processing/reprojection.rs]: 7af24a5

**Review by michaelmattig — APPROVED** [review]

**Comment by michaelmattig** [comment]
bors merge

**Comment by michaelmattig** [comment]
bors merge

**Review by michaelmattig — APPROVED** [review]

**Comment by michaelmattig** [comment]
bors merge

**Pull request merged**